scipy/stats/_distn_infrastructure.py

```diff
@@ -1186,8 +1186,14 @@ def moment(self, n, *args, **kwds):
 
         """
         args, loc, scale = self._parse_args(*args, **kwds)
-        if not (self._argcheck(*args) and (scale > 0)):
-            return nan
+        arrs = tuple(map(np.asarray, [*args, loc, scale]))
```

Comment:
This first assignment appears to be unused?

Reply:
`*args`? I think I did it because `loc` and `scale` parameters need to be of the same shape as inputs (that are present in `args`) otherwise the output isn't consistent with input shapes...

Reply:
The first line is not needed.
```
arrs = tuple(map(np.asarray, [*args, loc, scale]))
arrs = tuple(map(np.atleast_1d, [*args, loc, scale]))
```

Doing this causes the return value to always be an array if I'm understanding the code correctly.

Reply:
atleast_1d receives a list of arrays already, so the tuple(map(...)) dance can be avoided.

Reply:
Done. Thanks for the review!

Reply:
Do you still need the `arrs = tuple(map(np.asarray, [*args, loc, scale]))` even? 
`arrs = np.broadcast_arrays(*(*args, loc, scale))` accepts inputs that are not already arrays.